More dynamicDowncast<Document> and adjacent dynamicDowncast<> adoption

annevk opened the pull request:
#### 10a815cfebf7c0bb1cbae781a57de67cd96edeee
<pre>
More dynamicDowncast&lt;Document&gt; and adjacent dynamicDowncast&lt;&gt; adoption
<a href="https://bugs.webkit.org/show_bug.cgi?id=268090">https://bugs.webkit.org/show_bug.cgi?id=268090</a>

Reviewed by Chris Dumez.

The result of grepping for is&lt;Document&gt; and then replacing downcast&lt;&gt;
calls in the returned files. Quite a bit of Web Inspector code this
time around.

Also increase RefPtr usage.

* Source/WebCore/Modules/cookie-store/CookieStore.cpp:
(WebCore::CookieStore::stop):
* Source/WebCore/inspector/InspectorInstrumentation.cpp:
(WebCore::frameForScriptExecutionContext):
(WebCore::InspectorInstrumentation::instrumentingAgents):
* Source/WebCore/inspector/agents/InspectorAnimationAgent.cpp:
(WebCore::InspectorAnimationAgent::enable):
(WebCore::InspectorAnimationAgent::requestEffectTarget):
(WebCore::InspectorAnimationAgent::willApplyKeyframeEffect):
(WebCore::InspectorAnimationAgent::frameNavigated):
* Source/WebCore/inspector/agents/InspectorCSSAgent.cpp:
(WebCore::InspectorCSSAgent::getMatchedStylesForNode):
(WebCore::InspectorCSSAgent::getInlineStylesForNode):
(WebCore::InspectorCSSAgent::layoutFlagsForNode):
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
(WebCore::InspectorDOMAgent::unbind):
(WebCore::InspectorDOMAgent::assertDocument):
(WebCore::InspectorDOMAgent::assertElement):
(WebCore::InspectorDOMAgent::assertEditableNode):
(WebCore::InspectorDOMAgent::assertEditableElement):
(WebCore::InspectorDOMAgent::querySelector):
(WebCore::InspectorDOMAgent::querySelectorAll):
(WebCore::InspectorDOMAgent::insertAdjacentHTML):
(WebCore::InspectorDOMAgent::setNodeValue):
(WebCore::InspectorDOMAgent::highlightSelector):
(WebCore::InspectorDOMAgent::buildObjectForNode):
(WebCore::InspectorDOMAgent::buildObjectForEventListener):
(WebCore::InspectorDOMAgent::buildObjectForAccessibilityProperties):
(WebCore::containsOnlyASCIIWhitespace):
(WebCore::InspectorDOMAgent::innerParentNode):
(WebCore::InspectorDOMAgent::didAddEventListener):
(WebCore::InspectorDOMAgent::willRemoveEventListener):
(WebCore::InspectorDOMAgent::nodeForPath):
* Source/WebCore/inspector/agents/InspectorNetworkAgent.cpp:
(WebCore::InspectorNetworkAgent::didReceiveResponse):
(WebCore::InspectorNetworkAgent::enable):
(WebCore::InspectorNetworkAgent::resolveWebSocket):
* Source/WebCore/inspector/agents/InspectorTimelineAgent.cpp:
(WebCore::frame):
* Source/WebCore/inspector/agents/page/PageNetworkAgent.cpp:

Canonical link: <a href="https://commits.webkit.org/273601@main">https://commits.webkit.org/273601@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/37978cbd879569543288e24c2ef40f6f10ec7919

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/35866 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/14809 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/38056 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/38589 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/32267 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/37085 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/17224 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/11835 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31056 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/36420 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/12523 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/31869 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/11001 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/10996 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/32040 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/39839 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/32588 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/32364 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/36956 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/11178 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/9080 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/35046 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/12928 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8195 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/11688 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/12009 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/inspector/InspectorInstrumentation.cpp:
```diff
@@ -1371,10 +1369,10 @@ InstrumentingAgents& InspectorInstrumentation::instrumentingAgents(Page& page)
-    if (is<WorkerOrWorkletGlobalScope>(context))
-        return &instrumentingAgents(downcast<WorkerOrWorkletGlobalScope>(context));
+    if (RefPtr document = dynamicDowncast<Document>(context))
+        return instrumentingAgents(document->page());
```
`document->protectedPage().get()`

cdumez commented on Source/WebCore/inspector/agents/InspectorAnimationAgent.cpp:
```diff
@@ -260,11 +260,11 @@ Protocol::ErrorStringOr<void> InspectorAnimationAgent::enable()
 
         // FIXME: <https://webkit.org/b/168475> Web Inspector: Correctly display iframe's WebSockets
-        auto* document = downcast<Document>(scriptExecutionContext);
         return document->page() == &m_inspectedPage;
```
Could do `return document && document->page() == &m_inspectedPage;` on one line.

cdumez commented on Source/WebCore/inspector/agents/InspectorAnimationAgent.cpp:
```diff
@@ -299,13 +299,11 @@ Protocol::ErrorStringOr<Ref<Protocol::DOM::Styleable>> InspectorAnimationAgent::
 
-    auto* effect = animation->effect();
-    if (!is<KeyframeEffect>(effect))
+    auto* keyframeEffect = dynamicDowncast<KeyframeEffect>(animation->effect());
```
Surprised you didn't use a RefPtr here.

cdumez commented on Source/WebCore/inspector/agents/InspectorAnimationAgent.cpp:
```diff
@@ -382,10 +380,11 @@ static bool isDelayed(const ComputedEffectTiming& computedTiming)
 {
     auto* animation = keyframeEffect.animation();
-    if (!is<StyleOriginatedAnimation>(animation))
+    auto* styleOriginatedAnimation = dynamicDowncast<StyleOriginatedAnimation>(animation);
```
Surprised you didn't use a RefPtr here.

cdumez commented on Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
```diff
@@ -243,10 +243,10 @@ class EventFiredCallback final : public EventListener {
     void handleEvent(ScriptExecutionContext&, Event& event) final
     {
-        if (!is<Node>(event.target()) || m_domAgent.m_dispatchedEvents.contains(&event))
+        auto* node = dynamicDowncast<Node>(event.target());
```
A RefPtr would be nice.

cdumez commented on Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
```diff
@@ -427,13 +427,12 @@ void InspectorDOMAgent::unbind(Node& node)
-        Element& element = downcast<Element>(node);
-        if (ShadowRoot* root = element.shadowRoot())
+    if (RefPtr element = dynamicDowncast<Element>(node)) {
+        if (auto* root = element->shadowRoot())
```
A RefPtr would be nice.

cdumez commented on Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
```diff
@@ -427,13 +427,12 @@ void InspectorDOMAgent::unbind(Node& node)
+        if (auto* root = element->shadowRoot())
             unbind(*root);
-        if (PseudoElement* beforeElement = element.beforePseudoElement())
+        if (auto* beforeElement = element->beforePseudoElement())
```
A RefPtr would be nice.

cdumez commented on Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
```diff
@@ -427,13 +427,12 @@ void InspectorDOMAgent::unbind(Node& node)
+        if (auto* beforeElement = element->beforePseudoElement())
             unbind(*beforeElement);
-        if (PseudoElement* afterElement = element.afterPseudoElement())
+        if (auto* afterElement = element->afterPseudoElement())
```
A RefPtr would be nice.

cdumez commented on Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
```diff
@@ -459,31 +458,29 @@ Node* InspectorDOMAgent::assertNode(Protocol::ErrorString& errorString, Protocol
 Document* InspectorDOMAgent::assertDocument(Protocol::ErrorString& errorString, Protocol::DOM::NodeId nodeId)
 {
-    Node* node = assertNode(errorString, nodeId);
+    auto* node = assertNode(errorString, nodeId);
```
A RefPtr would be nice.

cdumez commented on Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
```diff
@@ -459,31 +458,29 @@ Node* InspectorDOMAgent::assertNode(Protocol::ErrorString& errorString, Protocol
 Element* InspectorDOMAgent::assertElement(Protocol::ErrorString& errorString, Protocol::DOM::NodeId nodeId)
 {
-    Node* node = assertNode(errorString, nodeId);
+    auto* node = assertNode(errorString, nodeId);
```
A RefPtr would be nice.

cdumez commented on Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
```diff
@@ -459,31 +458,29 @@ Node* InspectorDOMAgent::assertNode(Protocol::ErrorString& errorString, Protocol
 Node* InspectorDOMAgent::assertEditableNode(Protocol::ErrorString& errorString, Protocol::DOM::NodeId nodeId)
 {
-    Node* node = assertNode(errorString, nodeId);
+    auto* node = assertNode(errorString, nodeId);
```
A RefPtr would be nice.

cdumez commented on Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
```diff
@@ -499,14 +496,13 @@ Node* InspectorDOMAgent::assertEditableNode(Protocol::ErrorString& errorString,
 Element* InspectorDOMAgent::assertEditableElement(Protocol::ErrorString& errorString, Protocol::DOM::NodeId nodeId)
 {
-    Node* node = assertEditableNode(errorString, nodeId);
+    auto* node = assertEditableNode(errorString, nodeId);
```
A RefPtr would be nice.

cdumez commented on Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
```diff
@@ -636,13 +632,14 @@ Protocol::ErrorStringOr<std::optional<Protocol::DOM::NodeId>> InspectorDOMAgent:
     Protocol::ErrorString errorString;
 
-    Node* node = assertNode(errorString, nodeId);
+    auto* node = assertNode(errorString, nodeId);
```
A RefPtr would be nice.

cdumez commented on Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
```diff
@@ -661,13 +658,14 @@ Protocol::ErrorStringOr<Ref<JSON::ArrayOf<Protocol::DOM::NodeId>>> InspectorDOMA
     Protocol::ErrorString errorString;
 
-    Node* node = assertNode(errorString, nodeId);
+    auto* node = assertNode(errorString, nodeId);
```
a RefPtr would be nice.

cdumez commented on Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
```diff
@@ -931,14 +929,15 @@ Protocol::ErrorStringOr<void> InspectorDOMAgent::insertAdjacentHTML(Protocol::DO
     Protocol::ErrorString errorString;
 
-    Node* node = assertEditableNode(errorString, nodeId);
+    auto* node = assertEditableNode(errorString, nodeId);
```
a RefPtr would be nice.

cdumez commented on Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
```diff
@@ -948,14 +947,15 @@ Protocol::ErrorStringOr<void> InspectorDOMAgent::setNodeValue(Protocol::DOM::Nod
     Protocol::ErrorString errorString;
 
-    Node* node = assertEditableNode(errorString, nodeId);
+    auto* node = assertEditableNode(errorString, nodeId);
```
a RefPtr would be nice.

cdumez commented on Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
```diff
@@ -1958,57 +1957,51 @@ Ref<Protocol::DOM::Node> InspectorDOMAgent::buildObjectForNode(Node* node, int d
+    if (RefPtr element = dynamicDowncast<Element>(*node)) {
+        value->setAttributes(buildArrayForElementAttributes(element.get()));
+        if (RefPtr frameOwner = dynamicDowncast<HTMLFrameOwnerElement>(*element)) {
+            if (auto* document = frameOwner->contentDocument())
```
a RefPtr would be nice.

cdumez commented on Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
```diff
@@ -1958,57 +1957,51 @@ Ref<Protocol::DOM::Node> InspectorDOMAgent::buildObjectForNode(Node* node, int d
         }
 
-        if (ShadowRoot* root = element.shadowRoot()) {
+        if (auto* root = element->shadowRoot()) {
```
a RefPtr would be nice.

cdumez commented on Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
```diff
@@ -2076,11 +2069,10 @@ Ref<Protocol::DOM::EventListener> InspectorDOMAgent::buildObjectForEventListener
     String scriptID;
     if (auto* scriptListener = dynamicDowncast<JSEventListener>(eventListener.get()); scriptListener && scriptListener->isolatedWorld()) {
         Document* document = nullptr;
```
a RefPtr would be nice.

cdumez replied:
And then you can drop the `= nullptr`

cdumez commented on Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
```diff
@@ -2136,8 +2128,8 @@ Ref<Protocol::DOM::EventListener> InspectorDOMAgent::buildObjectForEventListener
         .release();
-    if (is<Node>(eventTarget))
-        value->setNodeId(pushNodePathToFrontend(&downcast<Node>(eventTarget)));
+    if (auto* node = dynamicDowncast<Node>(eventTarget))
```
a RefPtr would be nice.

cdumez commented on Source/WebCore/inspector/agents/InspectorNetworkAgent.cpp:
```diff
@@ -871,10 +871,10 @@ Protocol::ErrorStringOr<void> InspectorNetworkAgent::enable()
 
         for (auto* webSocket : activeWebSockets()) {
-            if (!is<Document>(webSocket->scriptExecutionContext()))
+            auto* document = dynamicDowncast<Document>(webSocket->scriptExecutionContext());
```
a RefPtr would be nice.

annevk committed:
More dynamicDowncast<Document> and adjacent dynamicDowncast<> adoption
https://bugs.webkit.org/show_bug.cgi?id=268090

Reviewed by Chris Dumez.

The result of grepping for is<Document> and then replacing downcast<>
calls in the returned files. Quite a bit of Web Inspector code this
time around.

Also increase RefPtr usage.

* Source/WebCore/Modules/cookie-store/CookieStore.cpp:
(WebCore::CookieStore::stop):
* Source/WebCore/inspector/InspectorInstrumentation.cpp:
(WebCore::frameForScriptExecutionContext):
(WebCore::InspectorInstrumentation::instrumentingAgents):
* Source/WebCore/inspector/agents/InspectorAnimationAgent.cpp:
(WebCore::InspectorAnimationAgent::enable):
(WebCore::InspectorAnimationAgent::requestEffectTarget):
(WebCore::InspectorAnimationAgent::willApplyKeyframeEffect):
(WebCore::InspectorAnimationAgent::frameNavigated):
* Source/WebCore/inspector/agents/InspectorCSSAgent.cpp:
(WebCore::InspectorCSSAgent::getMatchedStylesForNode):
(WebCore::InspectorCSSAgent::getInlineStylesForNode):
(WebCore::InspectorCSSAgent::layoutFlagsForNode):
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
(WebCore::InspectorDOMAgent::unbind):
(WebCore::InspectorDOMAgent::assertDocument):
(WebCore::InspectorDOMAgent::assertElement):
(WebCore::InspectorDOMAgent::assertEditableNode):
(WebCore::InspectorDOMAgent::assertEditableElement):
(WebCore::InspectorDOMAgent::querySelector):
(WebCore::InspectorDOMAgent::querySelectorAll):
(WebCore::InspectorDOMAgent::insertAdjacentHTML):
(WebCore::InspectorDOMAgent::setNodeValue):
(WebCore::InspectorDOMAgent::highlightSelector):
(WebCore::InspectorDOMAgent::buildObjectForNode):
(WebCore::InspectorDOMAgent::buildObjectForEventListener):
(WebCore::InspectorDOMAgent::buildObjectForAccessibilityProperties):
(WebCore::containsOnlyASCIIWhitespace):
(WebCore::InspectorDOMAgent::innerParentNode):
(WebCore::InspectorDOMAgent::didAddEventListener):
(WebCore::InspectorDOMAgent::willRemoveEventListener):
(WebCore::InspectorDOMAgent::nodeForPath):
* Source/WebCore/inspector/agents/InspectorNetworkAgent.cpp:
(WebCore::InspectorNetworkAgent::didReceiveResponse):
(WebCore::InspectorNetworkAgent::enable):
(WebCore::InspectorNetworkAgent::resolveWebSocket):
* Source/WebCore/inspector/agents/InspectorTimelineAgent.cpp:
(WebCore::frame):
* Source/WebCore/inspector/agents/page/PageNetworkAgent.cpp:

Canonical link: https://commits.webkit.org/273601@main
